jvcproj/__init__.py

```diff
@@ -0,0 +1,367 @@
+        """
+        self.logger = logging.getLogger(__name__)
+        self._sh=smarthome
+        self.host_port = (host, int(port))
+        self.gammaconf_dir = gammaconf_dir
+        self.logger.debug("Plugin '{}': configured for host: '{}'".format(self.get_fullname(), self.host_port))
+
+    def run(self):
+        """
+        Run method for the plugin - called once to start the plugins processing
+        """        
+        self.logger.debug("Plugin '{}': run method called".format(self.get_fullname()))
+        self.alive = True
+
+    def stop(self):
+        """
+        Stop method for the plugin
+        """
+        self.logger.debug("Plugin '{}': stop method called".format(self.get_fullname()))
+        self.alive = False
+
+    def parse_item(self, item):
+        """
+        Plugin's parse_item method. Is called for every item when the plugin is initialized.
+        
+        The plugin can, corresponding to its attribute keywords, decide what to do with
+        the item in future, like adding it to an internal array for future reference
+        :param item:    The item to process.
+        :return:        If the plugin needs to be informed of an items change you should return a call back function
+                        like the function update_item down below. An example when this is needed is the knx plugin
+                        where parse_item returns the update_item function when the attribute knx_send is found.
+                        This means that when the items value is about to be updated, the call back function is called
+                        with the item, caller, source and dest as arguments and in case of the knx plugin the value
+                        can be sent to the knx with a knx write function within the knx plugin.
+        """
+        # normal items
+        if self.has_iattr(item.conf, "jvcproj_cmd"):
+            self.logger.debug("Plugin '{}': Item '{}' with value '{}' found!"
+                              .format(self.get_fullname(), item, self.get_iattr_value(item.conf, 'jvcproj_cmd')))
+            return self.update_item
+        if self.has_iattr(item.conf, "jvcproj_gamma"):
+            self.logger.debug("Plugin '{}': Item '{}' with value '{}' found!"
+                              .format(self.get_fullname(), item, self.get_iattr_value(item.conf, 'jvcproj_gamma')))
+            return self.update_item
+
+    def parse_logic(self, logic):
+        pass
+
+    def update_item(self, item, caller=None, source=None, dest=None):
+        """
+        Write items values
+        
+        :param item: item to be updated towards the plugin
+        :param caller: if given it represents the callers name
+        :param source: if given it represents the source
+        :param dest: if given it represents the dest
+        """
+        if item():
+            if self.has_iattr(item.conf, 'jvcproj_cmd'):
+                if self.get_iattr_value(item.conf, 'jvcproj_cmd') == 'None':
```

Comment:
Why do you need a special 'None'  check? Is the `has_iattr()` check one line above not enough? Do you really want to set the attribute to value `"None"`?

Reply:
smartplugin.py returns True if item exists. But when no command is declared (jvcproj_cmd = ..) the returned value will be "None" or not? So you can debug a message to have a look in the item because no command is given there. That my intension.